### ISSUE TITLE
compile error when installing from opam

### ISSUE THREAD

**Issue by seliopou** [tategakibunko/jingoo#16]
Using the 4.02.3 switch and running `opam install jingoo`, I get the following output:

```
$ opam install jingoo
The following actions will be performed:
  - install jingoo 1.2.10

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default] https://opam.ocaml.org/archives/jingoo.1.2.10+opam.tar.gz downloaded

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of jingoo failed at "gmake".
Processing  1/1: [jingoo: ocamlfind remove]
#=== ERROR while installing jingoo.1.2.10 =====================================#
# opam-version 1.2.1
# os           freebsd
# command      gmake
# path         /home/vagrant/.opam/4.02.3/build/jingoo.1.2.10
# compiler     4.02.3
# exit-code    2
# env-file     /tmp/jingoo/jingoo-18644-e7966f.env
# stdout-file  /tmp/jingoo/jingoo-18644-e7966f.out
# stderr-file  /tmp/jingoo/jingoo-18644-e7966f.err
### stdout ###
# ocamlfind ocamlopt -g -package unix,dynlink,pcre,batteries -c jg_types.mli jg_types.ml jg_utils.ml jg_stub.mli jg_stub.ml  jg_parser.mli jg_parser.ml jg_lexer.ml  jg_runtime.ml jg_interp.ml  jg_template.mli jg_template.ml
# [...]
# ocamlfind ocamlopt -g -package unix,dynlink,pcre,batteries -o jingoo.cmxa -a 
# ocamlfind ocamlopt -g -thread -c 
# ocamlfind ocamlc -g -thread -c 
# ocamlfind ocamlopt -g -o compiler -linkpkg -package unix,dynlink,pcre,batteries jingoo.cmxa jg_cmdline.ml
# *** Error code 2
# 
# Stop.
# make[1]: stopped in /usr/home/vagrant/.opam/4.02.3/build/jingoo.1.2.10/src
# Makefile:2: recipe for target 'all' failed
### stderr ###
# Warning 3: deprecated: String.create
# [...]
# Use Bytes.create instead.
# File "jg_utils.ml", line 95, characters 14-27:
# Warning 3: deprecated: String.create
# Use Bytes.create instead.
# File "jg_cmdline.ml", line 1:
# Error: No implementations provided for the following modules:
#          Jg_template referenced from jg_cmdline.cmx
#          Jg_types referenced from jg_cmdline.cmx
# gmake: *** [all] Error 1



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  - install jingoo 1.2.10

No changes have been performed
```

**Comment by tategakibunko** [comment]
Maybe ocamlopt is not supported in your environment.

It's fixed by PR#17(if native compiler is not supprted, only bytecode library is build).

[Add a bytecode-only target#17](https://github.com/tategakibunko/jingoo/pull/17)

Could you try again by v1.2.11?

**Comment by seliopou** [comment]
I've traced the cause to recursive use of Makefiles. Specifically, `make` is hard-coded in all the Makefiles, rather than using the `$(MAKE) variable. I'm compiling on FreeBSD, which has a non-GNU version of make installed by default. I've installed GNU make as gmake, so the initial make invocation as gmake works fine but on "recursive" calls, it uses the non-GNU make instead.

I'll issue a pull request with a fix.
